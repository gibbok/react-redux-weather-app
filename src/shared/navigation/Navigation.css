.activeLink{
    color:green;
    font-weight: bold;
}