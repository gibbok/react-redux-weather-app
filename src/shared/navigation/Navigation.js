import React from 'react'
import { NavLink } from 'react-router-dom'

const Navigation = ({ onClick, id, title, tooltip, url }) => (
  <div onClick={onClick} alt={tooltip}>
    <NavLink
      to={url}
      exact
      activeStyle={{
        fontWeight: 'bold',
        color: 'red'
      }}
    >{title}</NavLink >
  </div >
)

export default Navigation
