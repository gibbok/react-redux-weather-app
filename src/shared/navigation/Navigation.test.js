import React from 'react'
import {shallow} from 'enzyme'
import Navigation from './Navigation'
import { NavLink } from 'react-router-dom'

describe('<Navigation />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Navigation
        onClick={'test'}
        id={1}
        title={'test title'}
        tooltip={'test tooltop'}
        url={'/map'}
       />
      )
    expect(wrapper.contains(
      <div onClick='test' alt='test tooltop'>
        <NavLink to='/map'
          exact
          activeStyle={{
            fontWeight: 'bold',
            color: 'red'
          }}
          activeClassName='active'>
          test title
        </NavLink>
      </div>
    )).toBe(true)
  })

  it('should should render one <NavLink /> component', () => {
    const wrapper = shallow(
      <Navigation
        onClick={'test'}
        id={1}
        title={'test title'}
        tooltip={'test tooltop'}
        url={'/map'}
     />
    )
    expect(wrapper.find(NavLink).length).toBe(1)
  })
})
