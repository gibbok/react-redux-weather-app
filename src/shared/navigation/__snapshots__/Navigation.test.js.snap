// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render with snapshot 1`] = `
<div
  alt="test tooltop"
  onClick="test"
>
  <NavLink
    activeClassName="active"
    activeStyle={
      Object {
        "color": "red",
        "fontWeight": "bold",
      }
    }
    exact={true}
    to="/map"
  >
    test title
  </NavLink>
</div>
`;
